tools/clang/test/HLSLFileCheck/passes/hl/hl_matrix_lower/dont_crash_on_invalid_cast.hlsl

```diff
@@ -0,0 +1,34 @@
+// RUN: %dxc -T cs_6_0 %s | xfail
+
+// The HL matrix lowering pass can sometimes throw an exception
+// due to an invalid LLVM-level cast<Ty> call.  Make sure that
+// propagates out to a user-level error.
+
+// Note: There is still a bug in the compiler here.  Not all matrix
+// lowerings are covered by the pass.
+
+// (I would like to use a FileCheck %s but that does not work.
```

Comment:
This can be tested better with the 'xfailcat' from #6711

Reply:
You can use FileCheck, as I did here: https://github.com/microsoft/DirectXShaderCompiler/pull/6556
Change it to:
```
// RUN: not %dxc -T cs_6_0 %s 2>&1 | FileCheck %s
```

Reply:
Can we make this an IR test instead to run via lit using `dxopt`?

Reply:
Changed to a lit-based test.